services/static.js

```diff
@@ -0,0 +1,245 @@
+'use strict';
+
+var polyline = require('@mapbox/polyline');
+var v = require('./service-helpers/validator');
+var createServiceFactory = require('./service-helpers/create-service-factory');
+var pick = require('./service-helpers/pick');
+
+/**
+ * Static API service.
+ */
+var Static = {};
+
+/**
+ * Get a static map image.
+ *
+ * See [the public documentation](https://www.mapbox.com/api-documentation/#retrieve-a-static-map-from-a-style).
+ *
+ * **If you just want the URL for the static map image, create a request
+ * and get it's URL with `MapiRequest#url`.**
+ *
+ * @param {Object} config
+ * @param {string} config.ownerId - The owner of the map style.
+ * @param {string} config.styleId - The map's style ID.
+ * @param {number} config.width - Width of the image in pixels, between 1 and 1280.
+ * @param {number} config.height - Height of the image in pixels, between 1 and 1280.
+ * @param {[number, number]|'auto'} config.coordinates - `[longitude, latitude]`
+ *   for the center of image; or `'auto'` to fit the map within the bounds of
+ *   the overlay features.
+ * @param {number} config.zoom - Between 0 and 20.
+ * @param {number} [config.bearing] - Between 0 and 360.
+ * @param {number} [config.pitch] - Between 0 and 60.
+ * @param {Overlay|Array<Overlay>} [config.overlay] - One or more overlays.
+ *   Overlays should be in z-index order: the first in the array will be on the
+ *   bottom; the last will be on the top. Overlays are objects that match one
+ *   of the following types.
+ *   - [`SimpleMarkerOverlay`](#simplemarkeroverlay)
+ *   - [`CustomMarkerOverlay`](#custommarkeroverlay)
+ *   - [`PathOverlay`](#pathoverlay)
+ *   - [`GeoJsonOverlay`](#geojsonoverlay)
+ * @param {boolean} [config.highRes=false]
+ * @param {string} [config.insertOverlayBeforeLayer] - The ID of the style layer
+ *   that overlays should be inserted *before*.
+ * @param {boolean} [config.attribution=true] - Whether there is attribution
+ *   on the map image.
+ * @param {boolean} [config.logo=true] - Whether there is a Mapbox logo
+ *   on the map image.
+ * @return {MapiRequest}
+ */
+Static.getStaticImage = function(config) {
+  v.assertShape({
+    ownerId: v.required(v.string),
+    styleId: v.required(v.string),
+    width: v.required(v.number),
+    height: v.required(v.number),
+    coordinates: v.required(v.oneOfType(v.coordinates, v.oneOf('auto'))),
+    zoom: v.required(v.number),
```

Comment:
we can use range for a bunch of props here

Reply:
Good idea 👍